### PR TITLE
migration of button icon to use TS box version

### LINKED ISSUE #20161: Update ButtonIcon to use TS Box version
### Description
Currently, the `ButtonIcon` component is using the deprecated JavaScript version of the `Box` component as a base component. This is causing issues with the polymorphic `as` prop and blocking further TypeScript migration in the extension.

Here are some examples of components that are using new TypeScript `Box` type conventions
- [ui/components/component-library/text/text.tsx](https://github.com/MetaMask/metamask-extension/blob/edbc728014c7eca3a0327e8ae979d461d6f3f336/ui/components/component-library/text/text.tsx)
- [ui/components/component-library/avatar-base/avatar-base.tsx](https://github.com/MetaMask/metamask-extension/blob/edbc728014c7eca3a0327e8ae979d461d6f3f336/ui/components/component-library/avatar-base/avatar-base.tsx)

### Technical Details
- Update the `ButtonIcon` component to use the TS version of the `Box` component 
- Migrate any constants to enums
- Update any deprecated constants to enums 

### Acceptance Criteria
- The `ButtonIcon` component is using the TS version of the `Box` component 

### Difficulty: Intermediate
Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by mishramonalisha76**
## Explanation
Used the new TS box Version in ButtonIcon component 

* Fixes #20161

### Before

https://github.com/MetaMask/metamask-extension/assets/8112138/ec341e39-e2d8-4246-8309-67446773c3d4

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/bd6ae3fc-c695-49d9-af71-9ba740866723

## Manual Testing Steps

- Go to the storybook build on this branch
- Search `ButtonIcon` 
- Check docs and stories work and make sense

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by mishramonalisha76** [comment]
I have read the CLA Document and I hereby sign the CLA

File: ui/components/component-library/avatar-icon/avatar-icon.types.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export interface AvatarIconStyleUtilityProps
   /**
    * Props for the icon inside AvatarIcon. All Icon props can be used
    */
-  iconProps?: IconStyleUtilityProps;
+  iconProps?: IconProps<'span'>;
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-icon/avatar-icon.types.ts:31]: Small fix to `AvatarIcon` types

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Thanks for your contribution @mishramonalisha76! I've pushed some small updates to our type conventions and docs.

**Comment by mishramonalisha76** [comment]
> LGTM! Thanks for your contribution @mishramonalisha76! I've pushed some small updates to our type conventions and docs.

Thanks @georgewrmarshall

**Review by garrettbear — DISMISSED** [review]

**Review by Ameralameri — APPROVED** [review]
Approv

**Review by georgewrmarshall — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Review by garrettbear — APPROVED** [review]

**Pull request merged**